modules/core/src/main/scala/org/scalasteward/core/data/Version.scala

```diff
@@ -157,61 +149,34 @@ object Version {
           case _                               => 0
         }
     }
+    final case class Hash(value: String, startIndex: Int) extends Component
```

Comment:
Another thing that came mind: Should we add a `case` to the `Order[Version]` instance for comparing `Hash` with `Hash` by comparing their values lexicographically so that versions which are equal up to the hashes are not considered equal?

Reply:
For the comparison we are only comparing `alnumComponents` which does not contain `Hash`.
So a `case (h1: Hash, h2: Hash) => h1.value.compare(h2.value)` to `Order[Component]` doesn't get triggered. 
But if I get your idea correctly you want to handle this directly in `Order[Version]` ? Feel free to add what you think.

On the other hand does it make sense to compare hashes lexicographically, this does not have to have a relationship with the right order.

Reply:
I was somehow concerned that we consider two versions equal even if their string representations are different. But since we are using `alnumComponents` in `Order[Version]` this is already the case now. I think you're also right that comparing hashes lexicographically doesn't make much sense for our purpose.